scripts/core/adapter.yml

```diff
@@ -177,6 +177,50 @@ returns:
     - $X_RESULT_ERROR_OUT_OF_RESOURCES
     - $X_RESULT_ERROR_OUT_OF_HOST_MEMORY
 --- #--------------------------------------------------------------------------
+type: fptr_typedef
+desc: "Callback function that processes adapter log messages."
+name: $x_logger_callback_t
+return: void
+params:
+  - type: $x_adapter_handle_t
+    name: hAdapter
+    desc: "[in] handle of the adapter"
+  - type: const char*
+    name: message
+    desc: "[in] the log message set by the unified runtime adapter."
+  - type: $x_log_level_t
+    name: logLevel
+    desc: "[in] The log message type"
+  - type: void*
+    name: pUserData
+    desc: "[in][optional] pointer to user data to be passed to the callback."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Set a function callback for use by the adapters when logging messages."
+details:
+  - "Sets a function callback that will be used by the adapters when logging messages."
+  - "The callback will be called for every message logged by the adapters specified in `phAdapters`."
+  - "Sending log messages is optional and adapters are not required to implement this feature for any entrypoint."
+  - "The application can disable log messages at runtime by setting the environment variable `UR_DISABLE_LOGS`."
```

Comment:
We should also define what happens when this entry point is called more than once. Do we replace the callback or append it to a list of callbacks?

I think replacing the callback would be simpler to implement. If that turns out to be an issue we can extend it later.

Reply:
I updated the wording on the description for this case. Also allowed the callback to be nullptr in case the application wants to unset it.

I'm not sure what to do with the environment variable `UR_DISABLE_LOGS`.  With the logger already having its own env_vars, I think that this variable is a bit redundant (if the application wants to ignore all the logs they can just unset the callback). But would like to hear someone else's opinion. It's not hard to implement the env var if we still think it is relevant.